### ISSUE TITLE
Sentry causes app to hang on start

### ISSUE THREAD

**Issue by mgray88** [getsentry/sentry-cocoa#1683]
### Platform

iOS

### Installed

Swift Package Manager

### Version

7.10.0

### Steps to Reproduce

Reproducing this has been about 6 months of tinkering to figure it out, we could never reproduce it reliably (if at all) in a debug build. Turned out to be because Sentry isn't usually initialized in our debug builds, but once we set that up, we could reproduce it reliably

Initially saw #1462, but it doesn't seem to be quite the same issue

1. Sentry is initialized in `application(_, didFinishLaunchingWithOptions:)` with
```swift
        SentrySDK.start { options in
            options.dsn = "<redacted>"
            options.enabled = true
            options.environment = ApiClient.endpoint.url.host
            options.enableAutoSessionTracking = true
        }
```
2. Cold start the app
3. Swipe close fairly quickly (not inhumanly fast)
4. Reopen the app
5. It hangs
6. At this point I could attach the debugger and pause execution to view the thread stack frames

There may very well be some of our code that is complicit in instigating this hang, but as I'll post in the result section, the only threads with anything significant are Sentry and Firebase

### Expected Result

No app hang

### The stack frames at execution pause:
<img width="493" alt="image" src="https://user-images.githubusercontent.com/411747/157350685-669f5c36-12a3-4596-abd3-d4550477f3f7.png">

- `monitorCachedData` is paused in [SentryCrashCashedData.c, line 139](https://github.com/getsentry/sentry-cocoa/blob/6e20ea06950ea871b70b6a82e8b42248ee2176e5/Sources/SentryCrash/Recording/SentryCrashCachedData.c#L139): `sleep(pollintInterval);` where:
  - `pollintInterval = 60` (probably should be pollingInterval 😄 )
  - `quickPollCount = 0`
  - `g_pollingIntervalInSeconds = 60`
- Thread 6 `handleExceptions` is paused in [SentryCrashMonitor_MachException.c, line 258](https://github.com/getsentry/sentry-cocoa/blob/6e20ea06950ea871b70b6a82e8b42248ee2176e5/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_MachException.c#L258): `thread_suspend((thread_t)sentrycrashthread_self());`
- Thread 7 `handleExceptions` is paused in [SentryCrashMonitor_MachException.c, line 266](https://github.com/getsentry/sentry-cocoa/blob/6e20ea06950ea871b70b6a82e8b42248ee2176e5/Sources/SentryCrash/Recording/Monitors/SentryCrashMonitor_MachException.c#L266): `kern_return_t kr = mach_msg(&exceptionMessage.header, MACH_RCV_MSG, 0,...`
- In both thread 6 & 7
  - `eventID` is the same, and is also equal to `g_primaryEventID`
  - `g_secondaryEventID` is different
  - `exceptionMessage` is empty
- What's interesting is it hangs here forever if left alone
- Force closing and reopening comes back to the same state
- However it will timeout after about 60 seconds if it's not open at the time of timeout

If there's anything else I can provide, I can pretty reliably reproduce it now

### Reproduced with debug:
- I just managed to reproduce it with Sentry debug enabled, and caught it on first launch
- In this case the only relevant thread stack trace is the identical to `monitorCachedData` above
- [Sentry Debug Log.txt](https://github.com/getsentry/sentry-cocoa/files/8210881/Sentry.Debug.Log.txt)

**Comment by mgray88** [comment]
Hi @brustolin, I'll see what I can do

**Comment by mgray88** [comment]
It looks like this may have been a bug in our code based on a misunderstanding of how a different library worked. An async issue that cropped up only occasionally, but when it did it appeared as though the app were "hanging," when really it was just not doing anything. Pausing it at that time, I saw `sleep(...)` which led me to falsely believe that was the culprit. I assume those Sentry threads then are normal operation and not related to the issue we experienced
